app-near/src/server/typeform/handleAnswer.ts

```diff
@@ -106,3 +71,132 @@ export const handleAnswer = async (req: NextRequest): Promise<NextResponse> => {
+  const { isNeighborhoodResident, ...createQuery } = parsedAnswer;
+
+  // no survey name in hidden fields
+  const surveyName = parsedBody.form_response.hidden?.neighborhood;
+  if (!surveyName) {
+    return noSurveyNameProvidedResponse(formId);
+  }
+
+  // continue only if we are in phase 2
+  const survey = await getOneSurveyByName(surveyName);
+  if (!survey) {
+    return noSurveyFoundResponse(surveyName);
+  }
+
+  if (isNotInPhaseTwo(survey)) {
+    return notInPhaseSuSurveyResponse(surveyName);
+  }
+
+  // create su
+  const broadcastChannel = parsedBody.form_response.hidden?.broadcast_channel;
+
+  await createSu({ ...createQuery, broadcastChannel } as SuAnswer, surveyName);
+
+  return NextResponse.json({ message: "created" }, { status: 201 });
+};
+
+const isValidSignature = (req: NextRequest, body: string): boolean => {
+  const signature = req.headers.get("Typeform-Signature");
+  return verifySignature(signature, body);
+};
+
+const okResponse = (message: string): NextResponse<{ message: string }> =>
+  NextResponse.json({ message }, { status: 200 });
+
+const unknwonFormIdResponse = (formId?: string): NextResponse =>
+  NextResponse.json(
+    { error: "Unknown formid", details: formId },
+    { status: 404 },
+  );
+
+const unauthorizedResponse = (formId?: string): NextResponse =>
+  NextResponse.json(
+    { error: "Not authorized", details: formId },
+    { status: 401 },
+  );
+
+const noSurveyFoundResponse = (
+  surveyName?: string,
+): NextResponse<{ error: string }> => {
+  console.error("[whebhook typeform]", surveyName, "Survey not found");
+  return NextResponse.json(
+    { error: `Survey name ${surveyName} not found` },
+    { status: 404 },
+  );
+};
+
+const noSurveyNameProvidedResponse = (
+  formId?: string,
+): NextResponse<{ error: string }> => {
+  console.error("[whebhook typeform]", formId, "Survey name not found");
+  return NextResponse.json({ error: "Survey name not found" }, { status: 400 });
+};
+
+const referenceMappingNotFoundResponse = (formId?: string): NextResponse => {
+  console.error(
+    "[whebhook typeform]",
+    formId,
+    "ERROR: References mapping not found",
+  );
+  return NextResponse.json(
+    { error: "References mapping not found" },
+    { status: 400 },
+  );
+};
+
+const notInPhaseSuSurveyResponse = (
+  surveyName: string,
+): NextResponse<{ error: string }> => {
+  console.error(
+    "[whebhook typeform]",
+    surveyName,
+    `step ${SurveyPhase.STEP_2_SU_SURVERY} is over for ${surveyName}`,
+  );
+  return NextResponse.json(
+    {
+      error: `step ${SurveyPhase.STEP_2_SU_SURVERY} is over for ${surveyName}`,
+    },
+    { status: 200 },
+  );
+};
+
+const isUnder15 = (answers: Record<string, string | boolean>) => {
+  return !Object.values(AgeCategory).includes(
+    answers.ageCategory as AgeCategory,
+  );
+};
+
+const isPartOfNeighborhood = (answers: Record<string, string | boolean>) => {
+  return answers.isNeighborhoodResident !== true;
+};
```

Comment:
:red_circle: J'ai l'impression que la logique est inversée ici ?
```suggestion
const isPartOfNeighborhood = (answers: Record<string, string | boolean>) => {
  return answers.isNeighborhoodResident === true;
};
```

Reply:
La méthode était mal nommée. Ca te va si je renomme en `isNotPartOfNeighborhood`